[Text Analytics] Re-recording tests based on v2022-10-01-preview

joseharriaga opened the pull request:
Resolves #31823.

joseharriaga committed:
Re-recording tests on v2022-10-01-preview

joseharriaga commented on sdk/textanalytics/Azure.AI.TextAnalytics/tests/AnalyzeHealthcareEntitiesTests.cs:
```diff
@@ -54,6 +54,8 @@ public AnalyzeHealthcareEntitiesTests(bool isAsync, TextAnalyticsClientOptions.S
         };
 
         [RecordedTest]
+        // Temporarily setting max version to V2022-05-01 since V2022-10-01-preview does not support AAD yet.
```
I will create an issue to keep track of removing this ServiceVersion.Max attribute once support for AAD is enabled for version 2022-10-01-preview.

heaths replied:
Tip: use `// TODO: ...` because Visual Studio has a Tasks view that tracks these, and it's much easier to find all `TODO` comments this way rather than a random comment.

I also recommend, as we normally do, putting a link to the tracking bug so that you can also find all TODOs specific to a particular issue and fix them all.

joseharriaga commented on sdk/textanalytics/Azure.AI.TextAnalytics/tests/AnalyzeHealthcareEntitiesTests.cs:
```diff
@@ -496,7 +498,7 @@ private void ValidateBatchDocumentsResult(AnalyzeHealthcareEntitiesResultCollect
         {
             Assert.AreNotEqual(new DateTimeOffset(), operation.CreatedOn);
-            Assert.AreNotEqual(new DateTimeOffset(), operation.LastModified);
+            // Assert.AreNotEqual(new DateTimeOffset(), operation.LastModified);
```
I will create an issue to keep track of this one. For now, the new 2022-10-01-preview version has a regression with this field that the service team is taking care of.

azure-sdk commented:
**API change check**

API changes are not detected in this pull request.

pallavit commented:
/cc: @kinelski as FYI

heaths commented on sdk/textanalytics/Azure.AI.TextAnalytics/tests/AnalyzeOperationTests.cs:
```diff
@@ -858,5 +858,15 @@ public void AnalyzeOperationSingleLabelClassifyActionNotSupported()
             NotSupportedException ex = Assert.ThrowsAsync<NotSupportedException>(async () => await client.StartAnalyzeActionsAsync(batchDocuments, batchActions));
             Assert.AreEqual("SingleLabelClassifyAction is not available in API version v3.1. Use service API version 2022-05-01 or newer.", ex.Message);
         }
+        private void ValidateOperationProperties(AnalyzeActionsOperation operation)
```
Nit: blank line between. You should be getting intellisense errors on these.

heaths reviewed (approved):
I recommend changing the comment to better track and later find/fix all the instances. Also, did you spot check recordings to make sure nothing unexpected happened?

joseharriaga committed:
Adding TODOs

joseharriaga commented:
@heaths I changed the comments to TODOs and added links to the issues that I created on GitHub. To answer your other question: Yes, I did take a look at the new recordings, and all the differences were within expectations.
